Remove honeycode from v1 (#5257)
aws/aws-sdk-go: CHANGELOG_PENDING.md

### SDK Features
* `service/honeycode`: Remove Honeycode

>   * This change removes the Honeycode service, since it will be deprecated.

Reviewer: ```suggestion
  * This change removes the Honeycode service, which has been deprecated.
```
